pkg/controller/configmapcainjector/configmapcainjector_controller.go

```diff
@@ -0,0 +1,165 @@
+}
+
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("configmap-trust-bundle-injector-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	proxyAnnotation := map[string]string{
+		names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION: "true",
+	}
+	cfpMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: proxyAnnotation,
+		},
+	}
+	// The events fire for changes/createion of the trusted-ca-bundle and any configmaps with the
+	//"config.openshift.io/inject-trusted-cabundle" annotation
+	pred := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.MetaNew.GetAnnotations()[names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION] == "true" ||
+				(e.MetaNew.GetName() == names.TRUST_BUNDLE_CONFIGMAP && e.MetaNew.GetNamespace() == names.TRUST_BUNDLE_CONFIGMAP_NS)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Meta.GetAnnotations()[names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION] == "true" ||
+				(e.Meta.GetName() == names.TRUST_BUNDLE_CONFIGMAP && e.Meta.GetNamespace() == names.TRUST_BUNDLE_CONFIGMAP_NS)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return e.Meta.GetAnnotations()[names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION] == "true" ||
+				(e.Meta.GetName() == names.TRUST_BUNDLE_CONFIGMAP && e.Meta.GetNamespace() == names.TRUST_BUNDLE_CONFIGMAP_NS)
+		},
+	}
+
+	err = c.Watch(&source.Kind{Type: cfpMap}, &handler.EnqueueRequestForObject{}, pred)
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileConfigMapInjector{}
+
+type ReconcileConfigMapInjector struct {
+	client client.Client
+	scheme *runtime.Scheme
+	status *statusmanager.StatusManager
+}
+
+func (r *ReconcileConfigMapInjector) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Printf("Reconciling update for ca-certs from %s/%s\n", request.Name, request.Namespace)
+
+	combinedCAbundleConfigMap := &corev1.ConfigMap{}
+	combinedCAbundleConfigMapName := types.NamespacedName{
+		Namespace: names.TRUST_BUNDLE_CONFIGMAP_NS,
+		Name:      names.TRUST_BUNDLE_CONFIGMAP,
+	}
+	err := r.client.Get(context.TODO(), combinedCAbundleConfigMapName, combinedCAbundleConfigMap)
+	if err != nil {
+		log.Println(err)
+		return reconcile.Result{}, err
+	}
+	combinedCAbundleData, err := r.validateTrustBundleConfigMap(combinedCAbundleConfigMap)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	//build a list of configMaps
+	configMapsToChange := []corev1.ConfigMap{}
+
+	//The trusted-ca-bundle changed
+	if request.Name == names.TRUST_BUNDLE_CONFIGMAP && request.Namespace == names.TRUST_BUNDLE_CONFIGMAP_NS {
+
+		listOfConfigMaps := &corev1.ConfigMapList{}
+		err = r.client.List(context.TODO(), &client.ListOptions{}, listOfConfigMaps)
+		if err != nil {
+			return reconcile.Result{}, nil
```

Comment:
Include error context, i.e. "Failed to list..."

Reply:
Use a list options [field selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/) to filter results by `names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION == "true"`.

Reply:
@danalanerh for configmaps the only valid field selectors are meta.name and meta.namespace

Reply:
@JacobTanenbaum ack.